include/benchmark/benchmark_api.h

```diff
@@ -230,7 +230,7 @@ BENCHMARK_UNUSED static int stream_init_anchor = InitializeStreams();
 // expression from being optimized away by the compiler. This function is
 // intended to add little to no overhead.
 // See: https://youtu.be/nXaxk27zwlk?t=2441
-#if defined(__GNUC__)
+#if defined(__GNUC__) && !defined(__pnacl__) && !defined(EMSCRIPTEN)
```

Comment:
I don't think Compiler vendors will even give us builtins with better behavior than the inline assembly.

The results of testing the fallback `DoNotOptimize` aren't going to be outrageous, but they're going to be orders of magnitude more expensive than the nop assembly instructions.